rust/src/verify.rs

```diff
@@ -258,11 +264,61 @@ fn get_padding(spec: &ics23::InnerSpec, branch: i32) -> Result<Padding> {
     }
 }
 
+// left_branches_are_empty returns true if the padding bytes correspond to all empty children
+// on the left side of this branch, ie. it's a valid placeholder on a leftmost path
+fn left_branches_are_empty(
+    spec: &ics23::InnerSpec,
+    op: &ics23::InnerOp,
+    branch: i32,
+) -> Result<bool> {
+    if let Some(&idx) = spec.child_order.iter().find(|&&x| x == branch) {
+        // compare the prefix bytes with the appropriate number of empty children
+        let left_children = spec.child_order.len() - 1 - idx as usize;
+        if op.prefix.len() < left_children * spec.child_size as usize {
+            return Ok(false);
+        }
+        let actual_prefix = op.prefix.len() - left_children * spec.child_size as usize;
```

Comment:
I think you could use `.checked_sub().ok_or(SomeError)` and avoid the if statement above (handle it here), but this is fine